README.rst

```diff
@@ -83,6 +84,13 @@ Each of these options must appear first on the command line.
     a colon, and then the line of text that matched. If no files are provided,
     all files returned by ``git ls-files`` are scanned.
 
+``--scan-history``
+    Scans repository including all revisions. When a file contains a secret, the
+    matched text from the file being scanned will be written to stdout and the
+    script will exit with a non-zero RC. Each matched line will be written with
+    the name of the file that matched, a colon, the line number that matched,
```

Comment:
I added `path` as a prohibited pattern in the `git-secrets` repo to see how this would work.

```
git secrets --add path
git secrets --scan-history
```

The output format is actually the commit SHA followed by the file. e.g.:
`84c348197b2d105168f133be7d6e1a1c75743f83:git-secrets:41:. $(git --exec-path)/git-sh-setup`. This should be reflected in the docs.

I'm trying to think about if this is problematic for working around false-positive pattern matches because if you just run `--scan`, the output is as follows:

`git-secrets:43:. $(git --exec-path)/git-sh-setup`

False positives are filtered out of results using a regular expression: https://github.com/awslabs/git-secrets#ignoring-false-positives. The false positive regex has the ability to match on the entire line of matching text, including the filename, line number, and matched line. If we take this change, false positives will need to take the commit SHA into account as well.

It seems like we either need to think of a better way to match on allowed patterns against the output of a prohibited pattern match, or the docs need to be updated to reflect this change in behavior.